### PR TITLE
16636 Display text as tooltip for request action item if text is too long

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#16636

*Description of changes:*
- app version = 5.0.19
- restored tooltip on Request Action v-select
- made tooltip conditional on whether element has clipped
- calculated whether to show tooltip
- misc cleanup
- deleted obsolete getter

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "name-request",
-  "version": "5.0.16",
```
Comment (severinbeauvais) [inline package.json:3]: The current version is actually 5.0.17. I don't know why GH isn't showing that here.

Ref: https://github.com/bcgov/namerequest/blob/feature-way-of-navigating/package.json

File: src/components/new-request/search.vue [hunk 2]
```diff
@@ -9,48 +9,62 @@
     <v-row class="mt-6">
       <!-- Request Action -->
       <v-col cols="12" md="6" class="py-0">
```
Comment (severinbeauvais) [inline src/components/new-request/search.vue:11]: The change below looks cleaner if you go to ⚙️ and then hide whitespace changes.

File: src/components/new-request/search.vue [hunk 3]
```diff
@@ -9,48 +9,62 @@
...
-              :class="{ 'hide-me': item.disabled }"
-            >
-              <div class="font-weight-bold">{{ item.text }}</div>
-              <div>{{ item.subtext }}</div>
-            </v-list-item-content>
+          <template v-slot:activator="scope">
+            <div v-on="scope.on">
+              <v-select
+                id="search-type-options-select"
+                class="request-action-select"
```
Comment (severinbeauvais) [inline src/components/new-request/search.vue:22]: I restored the id (from the original code) even though it's applied to the HTML input element, which is pretty useless for us right now.

The class is used as a selector to find the inner selection. I tried using a ref here instead -- which is equivalent to selecting this item -- but then crawling its children to find the inner selection was messy/fragile, whereas `querySelector()` is simple and probably more robust.

File: src/components/new-request/search.vue [hunk 4]
```diff
@@ -9,48 +9,62 @@
...
+                  >
+                    <div class="font-weight-bold">{{ item.text }}</div>
+                    <div>{{ item.subtext }}</div>
+                  </v-list-item-content>
+                </template>
+              </v-select>
+            </div>
           </template>
-        </v-select>
+          <span>{{ request && request.text }}</span>
```
Comment (severinbeauvais) [inline src/components/new-request/search.vue:66]: In the original code, this was `getRequestText` but that didn't return the correct value anyway, so I deleted it.

Note that this v-tooltip is conditional (line 16).

File: src/components/new-request/search.vue [hunk 5]
```diff
@@ -189,6 +203,7 @@
     </v-row>
 
     <!-- Corporate Number checkbox, only for XPro Canadian locations -->
+    <!-- *** TODO: change this to "isCanadian" -->
```
Comment (severinbeauvais) [inline src/components/new-request/search.vue:206]: I'll do this in my next PR (part of this ticket, anyway).

File: package.json (hunk 1 above)
Comment (severinbeauvais) [reply package.json]: Obsolete issue.

**Comment by pwei1018** [comment]
Temporary Url for review: https://namerequest-dev--pr-647-hix22fnx.web.app

File: src/components/new-request/search.vue [hunk 6]
```diff
@@ -611,6 +627,15 @@ export default class Search extends Mixins(CommonMixin) {
   onRequestActionChange (request: RequestActionsI): void {
     this.request = request
 
+    // calculate whether to show tooltip
+    // (in next tick after DOM update)
+    Vue.nextTick(() => {
+      const el = document.querySelector('.request-action-select .v-select__selection') as any
+      const offsetWidth = el?.offsetWidth as number
+      const scrollWidth = el?.scrollWidth as number
+      this.showRequestActionTooltip = (offsetWidth < scrollWidth)
```
Comment (JazzarKarim) [inline src/components/new-request/search.vue:636]: This is a pretty cool solution.

**Review by JazzarKarim — APPROVED** [review]
LGTM! 👍

File: src/components/new-request/search.vue (hunk 6 above)
Comment (severinbeauvais) [reply src/components/new-request/search.vue:636]: It's also the _only_ solution, because the text clipping is done automatically by the browser.

**Pull request merged**